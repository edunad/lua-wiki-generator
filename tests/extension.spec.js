#!/usr/bin/env node
'use strict';

const fs = require('fs-extra');
const { resolve } = require('path');

const MDExtension = require('../src/md-generators/extension.js');
const LuaParser = require('../src/parser.js');

const EXTENSION_TEMPLATE_FILE = './__test_templates__/EXTENSION_TEMPLATE.md';
const BAD_TEMPLATE_FILE = './__test_templates__/BAD_TEMPLATE.md';

describe('MDExtension', () => {
    it('Successfully generates a MD file', async () => {
        const extensionTemplate = await fs.readFile(resolve(__dirname, EXTENSION_TEMPLATE_FILE), 'utf8');
        expect(extensionTemplate).not.toBe(undefined);

        const md = new MDExtension('./home', extensionTemplate);
        expect(md).not.toBe(undefined);

        const parser = new LuaParser(resolve(__dirname, `./__test_lua__/ias.extension_test.lua`));
        expect(parser).not.toBe(undefined);

        const data = await parser.parseLua();
        expect(data).not.toBe(undefined);
        expect(data.blocks).not.toBe(undefined);
        expect(data.blocks.length).toBe(1);

        const mdGen = md.generate(data.blocks[0]);
        expect(mdGen).toMatchSnapshot();
    });

    it('Handles unknown fields', async () => {
        const classTemplate = await fs.readFile(resolve(__dirname, BAD_TEMPLATE_FILE), 'utf8');

        const md = new MDExtension('./home', classTemplate);
        expect(md).not.toBe(undefined);

        const parser = new LuaParser(resolve(__dirname, `./__test_lua__/ias.extension_test.lua`));
        expect(parser).not.toBe(undefined);

        const data = await parser.parseLua();
        expect(data).not.toBe(undefined);
        expect(data.blocks).not.toBe(undefined);
        expect(data.blocks.length).toBe(1);

        const mdGen = md.generate(data.blocks[0]);
        expect(mdGen).toMatchSnapshot();
    });

    it('Handles missing fields', async () => {
        const classTemplate = await fs.readFile(resolve(__dirname, EXTENSION_TEMPLATE_FILE), 'utf8');

        const md = new MDExtension('./home', classTemplate);
        expect(md).not.toBe(undefined);

        const parser = new LuaParser(resolve(__dirname, `./__test_lua__/ias.bad_extension_test.lua`));
        expect(parser).not.toBe(undefined);

        const data = await parser.parseLua();
        expect(data).not.toBe(undefined);
        expect(data.blocks).not.toBe(undefined);
        expect(data.blocks.length).toBe(1);

        const mdGen = md.generate(data.blocks[0]);
        expect(mdGen).toMatchSnapshot();
    });
});
