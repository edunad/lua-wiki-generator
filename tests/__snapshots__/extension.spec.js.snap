// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDExtension Handles missing fields 1`] = `
"## ![unknown](.gitbook/assets/unknown.png) math.isNan

\`\`\`lua
booleanasdasdada math.isNan(val)
\`\`\`


------
## Returns

| Type   | Description |
| ------ | ----------: |
| [booleanasdasdada](./home/booleanasdasdada) | aaaaaaaaaaaaaaaaaaaaa |

"
`;

exports[`MDExtension Handles unknown fields 1`] = `
"##![Base](.gitbook/assets/base.png) $TITLES_B$

![$RANDOM$](.gitbook/assets/$RANDOM$.png)
$UNKNOWN_FIELD$
"
`;

exports[`MDExtension Successfully generates a MD file 1`] = `
"## ![client](.gitbook/assets/client.png) math.isNan

\`\`\`lua
boolean math.isNan(val)
\`\`\`

Returns true if the number is NaN (not a number)
------
## Parameters

| Type   | Name | Description | Optional |
| ------ | ---- | ----------- | -------: |
| number | val | No description |  |

------
## Returns

| Type   | Description |
| ------ | ----------: |
| boolean | If its NaN |

"
`;
